plugins/inputs/win_services/win_services.go

```diff
@@ -19,19 +20,20 @@ import (
 //go:embed sample.conf
 var sampleConfig string
 
-type ServiceErr struct {
+type ServiceError struct {
 	Message string
 	Service string
 	Err     error
 }
 
-func (e *ServiceErr) Error() string {
+func (e *ServiceError) Error() string {
 	return fmt.Sprintf("%s: %q: %v", e.Message, e.Service, e.Err)
 }
 
 func IsPermission(err error) bool {
-	if err, ok := err.(*ServiceErr); ok {
-		return os.IsPermission(err.Err)
+	var serviceErr *ServiceError
+	if errors.As(err, &serviceErr) {
+		return os.IsPermission(serviceErr.Err)
```

Comment:
Can we please also get rid of `os.IsPermission`, the [docu](https://pkg.go.dev/os#IsPermission) says:
> This function predates errors.Is. It only supports errors returned by the os package. New code should use errors.Is(err, fs.ErrPermission).

Reply:
Fixed